python/graphscope/analytical/app/java_app.py

```diff
@@ -0,0 +1,381 @@
+        splited = graph.template_str.split("<")
+        java_app_type_params = self.frag_param_str.split(",")
+        num_type_params = 0
+        if len(splited) != 2:
+            raise Exception(
+                "Unrecoginizable graph template str: {}".format(graph.template_str)
+            )
+        if splited[0] == "vineyard::ArrowFragment":
+            if self.java_app_type != "property":
+                logger.error("Expected property app")
+                return False
+            if len(java_app_type_params) != 1:
+                logger.error("Expected one type params.")
+                return False
+            num_type_params = 1
+        if splited[1] == "gs::ArrowProjectedFragment":
+            if self.java_app_type != "projected":
+                logger.error("Expected projected app")
+                return False
+            if len(java_app_type_params) != 4:
+                logger.error("Expected 4 type params")
+                return False
+            num_type_params = 4
+        graph_actual_type_params = splited[1][:-1].split(",")
+        for i in range(0, num_type_params):
+            graph_actual_type_param = graph_actual_type_params[i]
+            java_app_type_param = java_app_type_params[i]
+            if not _type_param_consistent(graph_actual_type_param, java_app_type_param):
+                return False
+        return True
+
+    def _pack_jar(self, full_jar_path):
+        garfile = InMemoryZip()
+        tmp_jar_file = open(full_jar_path, "rb")
+        jar_bytes = tmp_jar_file.read()
+        garfile.append("{}".format(full_jar_path.split("/")[-1]), jar_bytes)
+        return garfile
+
+    def signature(self):
+        s = hashlib.sha256()
+        s.update(f"{self.type}.{self.jar_path}.{self.java_app_class}".encode("utf-8"))
+        s.update(self.gar)
+        return s.hexdigest()
+
+    @property
+    def java_app_class(self):
+        return self._java_app_class
+
+    @property
+    def cpp_driver_class(self):
+        return self._cpp_driver_class
+
+    @property
+    def jar_path(self):
+        return self._full_jar_path
+
+    @property
+    def jar_name(self):
+        return self._jar_name
+
+    @property
+    def java_app_type(self):
+        return self._java_app_type
+
+    @property
+    def frag_param_str(self):
+        return self._frag_param_str
+
+    def __call__(self, graph: Graph, *args, **kwargs):
+        kwargs_extend = dict(app_class=self.java_app_class, **kwargs)
+        if not hasattr(graph, "graph_type"):
+            raise InvalidArgumentError("Missing graph_type attribute in graph object.")
+
+        if (
+            self.java_app_type == "projected"
+            and graph.graph_type == graph_def_pb2.ARROW_PROPERTY
+        ):
+            graph = graph._project_to_simple()
+        app_ = graph.session._wrapper(JavaAppDagNode(graph, self))
+        return app_(*args, **kwargs_extend)
+
+
+class JavaAppDagNode(AppDAGNode):
```

Comment:
ditto. maybe `AppDAGNode` is enough to support java app ?

Reply:
We need extend the kwargs with some java app related info, such as fragment name, jar name, to run a java app. 
Put these stuff in AppDAGNode may seem too strange?

Reply:
@sighingnow Any comments?